learnhubmk/app
# Pagination blogs 463

## Summary of Changes

- Implemented pagination for the blogs page.
- Added pagination and itemsPerPage dropdown to the ReusableTable component.
- Fixed an unreported pagination bug where the table displayed no data if the user was on a specific page (e.g., page 3) and either searched or changed itemsPerPage. The page value is now reset properly before sending requests to the backend.
- Updated the fetch mechanism in blogs page to use React Query for improved data management.
- Connected the search functionality on the blogs page to the backend API.
- Implemented filtering mechanism (filters by status)
- Updated actions button for improved reusability and better UI + fixed bug with overflow being hidden

## Encountered Issues

- **Pagination Fix:** The current solution involves using useEffect in components that use Pagination. While this resolves the issue, the logic could be encapsulated within the Pagination component for better maintainability. Refactoring the ReusableTable to improve readability may also be required as part of this work.

- **Session Management**: Tokens are not invalidated upon expiration, leading to unexpected behavior when performing authorized actions after the session has expired. Addressing this issue may involve changes to authentication logic across the application.


## Manual Test Steps

1. Login with content-manager role or admin role 

2. Navigate to the blogs page. (/content-panel/blogs)
- Test pagination by navigating between pages.
- Change the itemsPerPage setting and verify the table updates correctly.

3. Test Search Functionality:
- Perform searches using the search bar.
- Verify that the displayed results match the query and that pagination resets when a new search is performed.

4. Review Fetch Mechanism:
- Ensure data is fetched correctly using React Query and that no errors occur during API calls.

5. UI and Functionality:
- Check the itemsPerPage dropdown and actions menu for functionality and UI alignment with the rest of the table.
